dynamodb/scripts/service.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+if [ ! -f outputs.json ]; then
+   echo "outputs.json file not found!"
+   exit 1
+fi
+
+# Render Output
+name=$(jq -r '.[] | select(.OutputKey=="TableName")|.OutputValue' outputs.json)
+arn=$(jq -r '.[] | select(.OutputKey=="TableARN")|.OutputValue' outputs.json)
+
+cat > /run/secrets/output<<EOF
+services: {
+	admin: {
+		consumer: permissions: rules: [{
+			apiGroups: ["aws.acorn.io"]
+			verbs: ["dynamodb:*"]
+			resources: ["${arn}", "${arn}/*"]
+		}]
+		data: {
+			arn:  "${arn}"
+			name: "${name}"
+		}
+	}
+	readonly: {
+		name: "DynamoDB Reader"
+		consumer: permissions: rules: [{
+			apiGroups: ["aws.acorn.io"]
+			verbs: [
+				"dynamodb:BatchGetItem",
+				"dynamodb:ConditionCheckItem",
+				"dynamodb:GetItem",
+				"dynamodb:GetRecords",
+				"dynamodb:GetShardIterator",
+				"dynamodb:PartiQLSelect",
+				"dynamodb:Query",
+				"dynamodb:Scan",
+			]
+			resources: ["${arn}", "${arn}/*"]
+		}]
+		data: {
+			arn:  "${arn}"
+			name: "${name}"
+		}
+	}
+	writeonly: {
+		name: "DynamoDB Writer"
+		consumer: permissions: rules: [{
+			apiGroups: ["aws.acorn.io"]
+			verbs: [
+				"dynamodb:BatchWriteItem",
+				"dynamodb:PartiQLInsert",
+				"dynamodb:PartiQLUpdate",
+				"dynamodb:PutItem",
+				"dynamodb:UpdateItem",
+			]
+			resources: ["${arn}", "${arn}/*"]
```

Comment:
what are the ${arn}/* resources?

Reply:
The intent was to enable permissions for all [resource types](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazondynamodb.html#amazondynamodb-resources-for-iam-policies). Perhaps the syntax is wrong? Also, it might only be needed for admin I suppose. 

I'm generally unsure which permissions should be included with each role because there are so many.

Reply:
AWS has precanned policies, maybe check those for guidance.

Reply:
The permissions seem okay to me. What do you think?